File: src/main/java/com/github/lkqm/paper/model/Project.java
package com.github.lkqm.paper.model;

import lombok.Data;

import java.io.Serializable;

/**
 * 项目实体
 *
 * @author Mario Luo
 * @date 2019.01.19 10:48
 */
@Data
public class Project implements Serializable {

    /**
     * 项目Id
     */
    private String id;

    /**
     * 项目名称
     */
    private String name;

    /**
     * 项目描述
     */
    private String description;

    /**
     * 项目创建人
     */
    private String creator;

    /**
     * 项目创建时间
     */
    private Long createTime;

    /**
     * 项目更新时间
     */
    private Long updateTime;

    /**
     * 类型, 1: 上传文件, 2: 外部地址
     */
    private Integer type = 1;

    /**
     * 项目入口: index.html
     */
    private String entranceUri;

    /**
     * 是否已经上传
     */
    private Boolean uploaded;

    /**
     * 外部跳转地址
     */
    private String linkUrl;
}

File: src/main/kotlin/model/Project.kt
package com.github.lkqm.paper.model


/**
 * 项目包装
 */
data class ProjectDataWrapper (
    var projects: MutableList<Project>? = null
)

/**
 * 项目实体
 */
data class Project (

    /** 项目Id */
    var id: String? = null,

    /** 项目名称 */
    var name: String? = null,

    /** 项目描述 */
    var description: String? = null,

    /** 项目创建人 */
    var creator: String? = null,

    /** 项目创建时间 */
    var createTime: Long? = null,

    /** 项目更新时间 */
    var updateTime: Long? = null,

    /** 类型, 1: 上传文件, 2: 外部地址 */
    var type: Int? = 1,

    /** 项目入口: index.html */
    var entranceUri: String? = null,

    /** 是否已经上传 */
    var uploaded: Boolean? = null,

    /** 外部跳转地址 */
    var linkUrl: String? = null
)

/**
 * 项目类型
 */
enum class ProjectType(val code: Int, val message: String) {

    FILE(1, "文件"),
    LINK(2, "链接");

    companion object {
        fun of(code: Int): ProjectType? {
            for (one in values()) {
                if (one.code == code) {
                    return one
                }
            }
            return null
        }
    }
}